### PR TITLE
Update cheaper node heartbeats KEP

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/enhancements/issues/589

### CODE REVIEW

File: keps/sig-node/0009-node-heartbeat.md [hunk 1]
```diff
@@ -189,27 +189,23 @@ Once all the code changes are done, we will:
...
-     significantly increase the size of object Node.
-     Note that if we keep adding node conditions owned by other components, the
-     number of writes of Node object will go up. But that issue is separate from
-     that proposal.
+Based on experiments, unsuccessful attempt to enable `Server Side Apply` due to
+performance issues it was causing, in v1.17 release, we will:
+- reduce frequency of NodeStatus update to once every 5 minutes instead of
+  1 minute (note that if any condition changes, the update will happen within
+  next 10 seconds anyway, so we are only talking about periodic heartbeats)
+- reduce frequency of updating NodeStatuses in NodeProblemDetector also to
```
Comment (bgrant0607) [inline keps/sig-node/0009-node-heartbeat.md]: I assume that if the status changes, NPD would also make the change promptly

File: keps/sig-node/0009-node-heartbeat.md [hunk 2]
```diff
@@ -228,6 +224,20 @@ relying on frequent Node object updates. However, in non-managed solutions, cust
 will still be able to restore previous behavior by setting appropriate flag values.
 Thus, changing defaults to what we recommend is the path to go with.
 
+To evaluate the risk of further reduction of NodeStatus updates, I went through
+couple thousands of files mentioning `LastHeartbeatTime` and `k8s` on github to
+understand the potential of breaking some system. There are couple places that
+are incompatible with node leases, but they would already be broken by current
+1m frequency, e.g.:
```
Comment (bgrant0607) [inline keps/sig-node/0009-node-heartbeat.md]: The Mesos Framework is deprecated, but would be nice to file issues on the other repos, if possible, to let them know about this KEP.

File: keps/sig-node/0009-node-heartbeat.md (hunk 1 above)
Comment (wangzhen127) [inline keps/sig-node/0009-node-heartbeat.md]: How do you plan to change NPD configuration? Use the new flag added in https://github.com/kubernetes/node-problem-detector/pull/356?

Comment (wojtek-t) [reply keps/sig-node/0009-node-heartbeat.md]: Yes - clarified in the text.

Comment (wojtek-t) [reply keps/sig-node/0009-node-heartbeat.md]: Kind-of implementation detail, but yes. Though we need to double check if it always works as we want (but that's doable, I'm focusing on having KEP approved first).

File: keps/sig-node/0009-node-heartbeat.md (hunk 2 above)
Comment (wojtek-t) [reply keps/sig-node/0009-node-heartbeat.md]: Filed (couldn't file the one for mesos, as the repo is already in read-only mode).

File: keps/sig-node/0009-node-heartbeat.md [hunk 3]
```diff
@@ -189,27 +189,29 @@ Once all the code changes are done, we will:
...
-     that should be reflected in [NPD documentation][].
-   - Note that reducing frequency to 1 minute already gives us almost 6x improvement.
-     It seems more than enough for any foreseeable future assuming we won’t
-     significantly increase the size of object Node.
-     Note that if we keep adding node conditions owned by other components, the
-     number of writes of Node object will go up. But that issue is separate from
-     that proposal.
+Based on experiments, unsuccessful attempt to enable `Server Side Apply` due to
+performance issues it was causing, in v1.17 release, we will:
+- reduce frequency of NodeStatus update to once every 5 minutes instead of
```
Comment (derekwaynecarr) [inline keps/sig-node/0009-node-heartbeat.md]: This is fine with me.

**Pull request merged**